flows/knack/knack_banner.py

```diff
@@ -0,0 +1,127 @@
+
+docker_image = f"atddocker/atd-knack-banner:{current_environment}"
+
+# Retrieve the email configuration
+email_config = get_key_value(key="aws_email_config")
+environment_variables = get_key_value(key=f"atd_knack_banner_{current_environment}")
+
+
+# Retrieve the provider's data
+@task(
+    name="HR knack banner integration",
+    max_retries=2,
+    retry_delay=timedelta(minutes=5),
+    state_handlers=[handler],
+    slug="knack-banner",
+)
+def knack_banner_update_employees():
+    response = (
+        docker.from_env()
+        .containers.run(
+            image=docker_image,
+            working_dir="/app",
+            command=f"./atd-knack-banner/update_employees.py",
+            environment=environment_variables,
+            volumes=None,
+            remove=True,
+            detach=False,
+            stdout=True,
+        )
+        .decode("utf-8")
+    )
+    logger = prefect.context.get("logger")
+    logger.info(response)
+    return response
+
+
+# Configure email task
+email_task = EmailTask(
+    name="email_task",
+    subject="Knack HR updates from Banner",
+    email_to=environment_variables["EMAIL_ADDRESS"],
+    email_to_cc="chia.berry@austintexas.gov",
+    email_from=email_config["email_from"],
+    smtp_server=email_config["smtp_server"],
+    smtp_port=email_config["smtp_port"],
+    smtp_type=email_config["smtp_type"],
+    attachments=None,
+)
+
+
+@task(log_stdout=True, state_handlers=[handler])
+def format_email_body(flow_data):
+    flow_data_list = flow_data.split("\n")
+    info_list = []
+    # only email the logs that are labeled "INFO:root"
+    for line in flow_data_list:
+        if line[0:9] == "INFO:root":
+            info_list.append(line[10:] + "<br>")
+    return " ".join(info_list)
+
+
+with Flow(
+    f"atd_knack_banner_{current_environment}",
+    storage=GitHub(
+        repo="cityofaustin/atd-prefect",
+        path="flows/knack/knack_banner.py",
+        ref=current_environment.replace("staging", "main")
+    ),
+    run_config=UniversalRun(labels=[current_environment, "atd-data02"]),
+    result=PrefectResult(),
+    schedule=Schedule(clocks=[CronClock("45 13 * * *")])
+) as get_data_flow:
+    email_data = knack_banner_update_employees()
+
+
+with Flow(
+    f"send_hr_email_{current_environment}",
+    storage=GitHub(
+        repo="cityofaustin/atd-prefect",
+        path="flows/test/knack_banner.py",
+        ref=current_environment.replace("staging", "main"),  # The branch name
+    ),
+    run_config=UniversalRun(labels=["test", "atd-data02"]),
+) as send_email_flow:
+    get_data_flow_run_id = create_flow_run(flow_name=get_data_flow.name)
+    script_result = get_task_run_result(
+        get_data_flow_run_id, task_slug="knack-banner-copy"
+    )
+    formatted_data = format_email_body(script_result)
+    send_email_flow.chain(script_result, formatted_data, email_task(msg=formatted_data))
+
+
+if __name__ == "__main__":
+    send_email_flow.run()
```

Comment:
How does the `send_email_flow` flow get triggered? I see the scheduled flow and task running each day for `atd_knack_banner` but how does that talk to the second `send_email` flow? 

Very cool thanks for sharing with me!

Reply:
There's some magic Sergio wrote here in the github actions: https://github.com/cityofaustin/atd-prefect/blob/main/.github/workflows/prefect_helper.sh#L79